php/class-delivery.php

```diff
@@ -153,6 +153,49 @@ function ( $post ) {
 		}
 	}
 
+	/**
+	 * Maybe filter out Cloudinary URLs in post meta.
+	 *
+	 * @param null|bool $check      Whether to allow adding metadata for the given type.
+	 * @param int       $object_id  The ID of the object metadata is for.
+	 * @param string    $meta_key   The Metadata key.
+	 * @param mixed     $meta_value Metadata value.
+	 *
+	 * @return null|bool
+	 */
+	public function maybe_filter_out_metadata( $check, $object_id, $meta_key, $meta_value ) {
+
+		$current       = current_filter();
+		$is_serialized = false;
+		$is_object     = is_object( $meta_value );
+
+		if ( ! is_string( $meta_value ) ) {
+			$is_serialized = true;
+			$meta_value    = wp_json_encode( $meta_value );
+		}
+
+		list( $action, $object ) = explode( '_', $current );
+
+		$process_meta_value = $meta_value;
+
+		remove_filter( $current, array( $this, 'maybe_filter_out_metadata' ) );
+
+		$process_meta_value = $this->filter_out_cloudinary( $process_meta_value );
+
+		if ( $process_meta_value !== $meta_value ) {
+			if ( $is_serialized ) {
+				$meta_value = json_decode( wp_unslash( $process_meta_value ), ! $is_object );
+			} else {
+				$meta_value = $process_meta_value;
+			}
+			$check = call_user_func( "{$action}_{$object}_meta", $object_id, $meta_key, $meta_value );
+		}
+
+		add_filter( $current, array( $this, 'maybe_filter_out_metadata' ), 10, 4 );
+
```

Comment:
@pereirinha - instead of removing and adding the filter, perhaps we could use a static variable. Set it true when processing, then false after. And return immediately if its true.